### DISCUSSION TITLE
Can I detect draftMode client-side (with pages router)?

### DISCUSSION THREAD

**Discussion by bplmp** [vercel/next.js#55401] [Help] [answered]
### Summary

I know I can do the following to detect if draft mode is on server-side:

```js
export async function getStaticProps(context) {
  if (context.draftMode) {
    // draftMode is on
  }
}
```

Is there any way I can do the same client-side **using pages router**? I've tried checking to see if the `__prerender_bypass` cookie is present, but because it is a SameSite Lax cookie I don't think that is possible.

## Why I want to do this

I have two versions of my website, a production and a preview one. In the preview, I want to enable draftMode on by default, so that the user doesn't need to go to an `/api/preview` route to enable it. But I can't seem to do it, except by redirecting the user to `/api/preview` and then back to where they were, and for that I need to detect client-side if draftMode is already on.

**Comment by ijjk** [comment, marked as answer]
Hi, yes you can detect if you are in preview mode in pages via the `isPreview` field on the router, related docs here https://nextjs.org/docs/pages/api-reference/functions/use-router#router-object

**Reply by bplmp** [reply]
Thanks for the super quick reply!
